Cargo.toml

```diff
@@ -108,6 +119,21 @@ tcp = [
     "tokio/time",
 ]
 
+# Tracing layers
+layers = [
```

Comment:
why is the feature flag to enable `tracing` support called `layers`? i would expect it to be called `tracing` or something...

Reply:
> why is the feature flag to enable `tracing` support called `layers`? i would expect it to be called `tracing` or something...

I believe that is where I started - again from memory - you can't have a feature name that is congruent to a crate name - but I'm not a Rust expert

Reply:
> you can't have a feature name that is congruent to a crate name - but I'm not a Rust expert

An [optional dependency](https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies) is equivalent to a feature flag that only enables one dependency. I think that `tracing` should just be an optional dependency and the instrumentation points should all be flagged with `#[cfg(feature = "tracing")]`. If we do end up exposing other functionality, like the OpenTelemetry layer you've implemented, it should be under a separate feature flag.

Reply:
> the instrumentation points should all be flagged with #[cfg(feature = "tracing")]

What do you think of using two flags `debug` and `trace` or some such.  Where the use cases are distinct, Use `debug` when if you are investigating the integration of your app and hyper.
Use `trace` when you are investigating hyper internals.
The debug spans would then be constrained to around only public functions.  

Not sure if that type of distinction can work out?


> If we do end up exposing other functionality, like the OpenTelemetry layer you've implemented, it should be under a separate feature flag.

Agreed, or probably better to have a separate crate